presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -331,10 +331,20 @@ public PlanWithProperties visitAggregation(AggregationNode node, StreamPreferred
             PlanWithProperties child = planAndEnforce(node.getSource(), childRequirements, childRequirements);
 
             List<VariableReferenceExpression> preGroupedSymbols = ImmutableList.of();
-            if (!LocalProperties.match(child.getProperties().getLocalProperties(), LocalProperties.grouped(groupingKeys)).get(0).isPresent()) {
+            // SortedByKeys here is already the longest prefix of the sorted-by columns that is a subset of the grouped-by columns
+            List<LocalProperty<VariableReferenceExpression>> sortedByKeys = child.getProperties().getLocalProperties();
+            if (!LocalProperties.match(sortedByKeys, LocalProperties.grouped(groupingKeys)).get(0).isPresent()) {
                 // !isPresent() indicates the property was satisfied completely
                 preGroupedSymbols = groupingKeys;
```

Comment:
I think we also need to check the streaming partition columns and make sure it is partitioned by the group columns across splits. IIRC, I didn't add this check in my prototype because it was implicitly checked. If the streaming aggregation doesn't match the group columns, a local exchange will be added below the aggregation and the local property will not match. We'd better document this assumption in the code comment.

However, the merge join planning has to check the partitioning columns across splits and ensure colocated join is planned. Only checking the local property will not work for merge join. (This is not related to this PR. But need to fix in the merge join planning code).